core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -382,6 +382,12 @@ abstract class AbstractFetcherThread(name: String,
                     "that the partition is being moved")
                   partitionsWithError += topicPartition
 
+                case Errors.UNKNOWN_TOPIC_OR_PARTITION =>
+                  warn(s"Receiving ${Errors.UNKNOWN_TOPIC_OR_PARTITION} from the leader for partition $topicPartition. " +
+                       s"This can happen transiently if the partition is being created or deleted. " +
+                       s"However, this is unexpected if it sustains.")
```

Comment:
Thanks for the update. I think the following reads a bit better and is a bit more consistent with the other messages:

```scala
s"Received ${Errors.UNKNOWN_TOPIC_OR_PARTITION} from the leader for partition $topicPartition. " +
"This error may be returned transiently when the partition is being created or deleted, but it is not " +
"expected to persist")
```

Reply:
@ijuma thanks for the update -- let me know if further changes needed.